predicators/nsrt_learning/strips_learning/pnad_search_learner.py

```diff
@@ -68,10 +69,16 @@ def _append_new_pnad_and_keep_effects(
         Return the final set of all PNADs (current, newly-created, and
         ones with keep effects).
         """
+        start_time = time.perf_counter()
         new_pnads = current_pnads + [new_pnad]
         # We first repartition data and ensure delete and ignore
         # effects for the newly-created PNAD are correct.
         new_pnads = self._learner.recompute_pnads_from_effects(new_pnads)
+        # For certain predicates, the above call can eat up significant
+        # time and memory. Thus, we check here if we exceed the preset
+        # timeout.
+        if time.perf_counter() - start_time > CFG.pnad_search_timeout:
```

Comment:
should this be `self._timeout`? or what is the relationship between `self._timeout` and `CFG.pnad_search_timeout`?